### ISSUE TITLE
Malformed principal not detected

### ISSUE THREAD

**Issue by gbudge** [aws-cloudformation/cfn-lint#4287]
### CloudFormation Lint Version

1.40.4

### What operating system are you using?

Mac

### Describe the bug

Malformed ARN not detected by cfn-lint, but is detected by CloudFormation.

CloudFormation Error:

_Resource handler returned message: "Invalid principal in policy: "AWS":"arn:aws::iam::<redacted>:root" (Service: Iam, Status Code: 400, Request ID: <redacted>) (SDK Attempt Count: 1)" (RequestToken: <redacted>, HandlerErrorCode: InvalidRequest)_

### Expected behavior

cfn-lint identifies the malformed ARN and raised an _**Invalid principal in policy**_ error.

### Reproduction template

```yaml
---
AWSTemplateFormatVersion: '2010-09-09'
Description: demo

Resources:
  Role:
    Type: AWS::IAM::Role
    Properties:
      RoleName: SomeRoleName
      Description: Role for something
      AssumeRolePolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action:
              - sts:AssumeRole
            Principal:
              AWS: !Sub 'arn:${AWS::Partition}::iam::${AWS::AccountId}:root'
```
